funcx_sdk/funcx/sdk/client.py

```diff
@@ -55,12 +56,13 @@ def __init__(
         fx_authorizer=None,
         search_authorizer=None,
         openid_authorizer=None,
-        funcx_service_address="https://api2.funcx.org/v2",
+        funcx_service_address=None,
         check_endpoint_version=False,
         asynchronous=False,
         loop=None,
-        results_ws_uri="wss://api2.funcx.org/ws/v2/",
+        results_ws_uri=None,
         use_offprocess_checker=True,
+        environment=None,
```

Comment:
This needs documentation to clarify that it only accepts "production" or "dev" or None.

Reply:
I was intentionally omitting this from public documentation for now.

In the case of globus-sdk, we cover many parameters with this in our documentation:
> All other parameters are for internal use and should be ignored.

I'd like to take a similar direction here. Users should not be passing values for `environment`.

---

Somewhat related: FuncXClient currently passes `environment="funcx"` to `BaseClient.__init__`. That's meaningless -- at best it does nothing and at worst it could have harmful side-effects. Because somebody "reached in" and tried passing a value that was intentionally documented as not being for their use.

I'm not sure what the intent was behind that line, but I don't want to be in a position of supporting users passing values for `environment`. They should not touch it.

Reply:
My prior comment has been bothering me all weekend. The relationship between the misuse of `environment` as a globus-sdk parameter doesn't quite support not documenting it in funcx.

I think my conclusion (because I don't think anything I said is wrong on its own merits) is that we should document `environment` as a parameter as "for internal use only" and not specify its type or values. I'll make a change to do just this.

Reply:
Getting into it to do the edits, I arrived at this change: 5a689ec

I think that's good, so I'll mark resolved. Let me know if there's an issue with this.